code/modules/recycling/conveyor2.dm

```diff
@@ -138,24 +136,34 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 	use_power(6)
 
 	//get the first 30 items in contents
-	affecting = list()
-	var/i = 0
 	var/list/items = loc.contents - src
-	for(var/item in items)
-		i++ // we're sure it's a real target to move at this point
-		if(i >= MAX_CONVEYOR_ITEMS_MOVE)
-			break
-		affecting.Add(item)
-
+	if(!LAZYLEN(items))//Dont do anything at all if theres nothing there but the conveyor
+		return
+	var/list/affecting = list()
+	if(length(items) > MAX_CONVEYOR_ITEMS_MOVE)
+		affecting = items.Copy(1, MAX_CONVEYOR_ITEMS_MOVE + 1)//Lists start at 1 lol
+	else
+		affecting = items
 	conveying = TRUE
-	addtimer(CALLBACK(src, .proc/convey, affecting), 1)
+
+	addtimer(CALLBACK(src, .proc/convey, affecting), 1)//Movement effect
 
 /obj/machinery/conveyor/proc/convey(list/affecting)
-	for(var/atom/movable/A in affecting)
+	for(var/O in affecting)
+		if(!ismovable(O))	//This is like a third faster than for(var/atom/movable in affecting)
+			continue
+		var/atom/movable/A = O
+		//Give this a chance to yield if the server is busy
+		stoplag()
 		if(!QDELETED(A) && (A.loc == loc))
```

Comment:
I think better invert this `if(!QDELETED(A) && (A.loc == loc))` to 
```
if(QDELETED(A) || (A.loc != loc))
	continue
```

Reply:
👍